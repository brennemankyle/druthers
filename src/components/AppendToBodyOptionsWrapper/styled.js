import styled from "@emotion/styled";
import { StyledAppendToBodyOptionsWrapper as RawStyledAppendToBodyOptionsWrapper } from "./AppendToBodyOptionsWrapper";

const StyledAppendToBodyOptionsWrapper = styled(
  RawStyledAppendToBodyOptionsWrapper
)`
  position: absolute;
  background-color: ${props => props.styles_colors_background};
  width: ${props => props.styles_width}px;

  ${props =>
    props.placeOptionsAbove
      ? `bottom: ${window.innerHeight - props.parentRect.y}px;
      left: ${props.parentRect.x}px;`
      : `top: ${props.parentRect.y + props.parentRect.height}px;
      left: ${props.parentRect.x}px;`}
  ${props => props.styles_rightToLeft && `transform: scaleX(-1);`}
  ${props =>
    !props.styles_hasOptions && !props.styles_multiple && `display: none;`}
`;

export { StyledAppendToBodyOptionsWrapper };
