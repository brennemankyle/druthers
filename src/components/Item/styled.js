import styled from "@emotion/styled";
import Item from "./Item";

const Selection = styled(Item)`
  border-radius: ${props => props.styles_borderRadius};
  align-items: center;
  justify-content: space-between;
  margin-top: calc(${props => props.styles_selection_margin} / 2);
  margin-bottom: calc(${props => props.styles_selection_margin} / 2);
  padding-top: ${props => props.styles_selection_paddingTop};
  padding-bottom: ${props => props.styles_selection_paddingBottom};
  padding-left: ${props => props.styles_selection_paddingLeft};
  padding-right: ${props =>
    props.removable ? "0" : props.styles_selection_paddingRight};

  & .remove:not(svg):not(path) {
    display: flex;
    border: none;
    border-radius: ${props => props.styles_borderRadius};
    margin-top: -${props => props.styles_selection_paddingTop};
    margin-bottom: -${props => props.styles_selection_paddingBottom};
    padding-top: ${props => props.styles_selection_paddingTop};
    padding-bottom: ${props => props.styles_selection_paddingBottom};
    padding-left: ${props => props.styles_selection_paddingLeft};
    padding-right: ${props => props.styles_selection_paddingRight};
    background-color: transparent;
    font-size: 1em;

    svg {
      fill: ${props => props.styles_colors_primary};
      width: ${props => props.styles_icon_width};
      height: ${props => props.styles_icon_width};
    }

    ${props =>
      props.styles_selectionHighlighted === props.item.value &&
      `background-color: ${props.styles_colors_warning};

      svg {
        fill: ${props.styles_colors_warningBold};
      }`}
  }

  ${props =>
    props.styles_multiple
      ? `display: inline-flex;
      background-color: ${props.styles_colors_secondary};
      margin-right: ${props.styles_selection_margin};`
      : `display: flex;`}
  ${props =>
    props.styles_rightToLeft &&
    `
    transform: scaleX(-1);
    flex-direction: row-reverse;
    padding-left: ${
      props.removable ? "0" : props.styles_selection_paddingRight
    };
    padding-right: ${props.styles_selection_paddingLeft};`}
`;

const Option = styled(Item)`
  padding-top: ${props => props.styles_option_paddingTop};
  padding-bottom: ${props => props.styles_option_paddingBottom};
  padding-right: ${props => props.styles_option_paddingRight};
  padding-left: ${props => props.styles_option_paddingLeft};
  &:first-of-type{
    border-top-left-radius: ${props => props.styles_borderRadius};
    border-top-right-radius: ${props => props.styles_borderRadius};
  }
  &:last-of-type{
    border-bottom-left-radius: ${props => props.styles_borderRadius};
    border-bottom-right-radius: ${props => props.styles_borderRadius};
  }
  z-index: 999;

  ${props =>
    props.styles_optionHighlighted != null &&
    props.styles_optionHighlighted === props.item.value &&
    `background-color: ${props.styles_colors_highlight};`}
  ${props =>
    props.styles_rightToLeft &&
    `transform: scaleX(-1);
    text-align: right;`}
  ${props =>
    props.item.childGroup &&
    props.item.value == null &&
    `color: ${props.styles_colors_secondary};
    font-size: .8em;`}
  ${props =>
    props.item.childGroup && props.item.value != null && `font-weight: bold;`}
  ${props =>
    props.item.group &&
    `margin-left: ${props.item.group.split(".").length * 2}em;`}
`;

export { Selection, Option };
