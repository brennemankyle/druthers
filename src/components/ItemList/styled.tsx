import React from "react";
import styled from "@emotion/styled";
import ItemList from "./ItemList";

const SelectionList = styled(ItemList)(
  (props) => `
  ${
    props.styles_multiple
      ? `display: inline;`
      : `display: inline-block;
      &:not(:empty) {
        width: 100%;
      }`
  }
  list-style-type: none;
  padding: 0;
  margin: 0;
  color: ${props.styles_colors_text};
`
);

const OptionList = styled(ItemList)(
  (props) => `
  color: ${props.styles_colors_text};
  list-style-type: none;
  padding: 0;
  margin: 0;
  margin-block-start: 0;
  margin-block-end: 0;
  border-color: ${props.styles_colors_secondary};
  border-width: ${props.styles_borderWidth};
  border-style: ${props.styles_borderStyle};
  border-radius: ${props.styles_borderRadius};

  .truncate-show, .truncate-hide {
    cursor: pointer;
  }
`
);

export { SelectionList, OptionList };
