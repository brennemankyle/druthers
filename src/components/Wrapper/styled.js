import styled from "@emotion/styled";
import RawWrapper from "./Wrapper";

const Wrapper = styled(RawWrapper)`
  font-size: ${props => props.styles_fontSize};
  ${props => props.styles_rightToLeft && `transform: scaleX(-1);`}
`;

const OptionsWrapper = styled(RawWrapper)`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 999;
  background-color: ${props => props.styles_colors_background};
  width: ${props => props.styles_width}px;
  ${props =>
    !props.styles_hasOptions && !props.styles_multiple && `display: none;`}
`;

export { Wrapper, OptionsWrapper };
