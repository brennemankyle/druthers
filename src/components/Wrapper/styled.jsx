import styled from "@emotion/styled";
import RawWrapper from "./Wrapper";

const Wrapper = styled(RawWrapper)(
  props => `
  font-size: ${props.styles_fontSize};
  ${props.styles_rightToLeft ? `transform: scaleX(-1);` : ``}
`
);

const OptionsWrapper = styled(RawWrapper)(
  props => `
  position: absolute;
  top: 0;
  left: 0;
  z-index: 999;
  background-color: ${props.styles_colors_background};
  width: ${props.styles_width}px;
  ${!props.styles_hasOptions && !props.styles_multiple ? `display: none;` : ``}
`
);

export { Wrapper, OptionsWrapper };
