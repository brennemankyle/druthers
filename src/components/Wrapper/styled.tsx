import React from "react";
import styled from "@emotion/styled";
import RawWrapper from "./Wrapper";

export const Wrapper = styled(RawWrapper)(
  (props) => `
  font-size: ${props.styles_fontSize};
  ${props.styles_rightToLeft ? `transform: scaleX(-1);` : ``}
`
);

export const OverlayOptionsWrapper = styled(RawWrapper)(
  (props) => `
  position: absolute;
  top: 0;
  left: 0;
  z-index: 999;
  background-color: ${props.styles_colors_background};
  width: ${props.styles_width}px;
  ${!props.styles_hasOptions && !props.styles_multiple ? `display: none;` : ``}
`
);

export const InPlaceOptionsWrapper = styled(Wrapper)`
  background-color: ${(props) => props.styles_colors_background};
  width: ${(props) => props.styles_width}px;
`;
