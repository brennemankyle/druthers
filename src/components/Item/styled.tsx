import React from "react";
import styled from "@emotion/styled";
import Item from "./Item";

const Selection = styled(Item)(
  (props) => `
  color: ${props.styles_colors_text}
  border-radius: ${props.styles_borderRadius};
  align-items: center;
  justify-content: space-between;
  margin-top: calc(${props.styles_selection_margin} / 2);
  margin-bottom: calc(${props.styles_selection_margin} / 2);
  padding-top: ${props.styles_selection_paddingTop};
  padding-bottom: ${props.styles_selection_paddingBottom};
  padding-left: ${props.styles_selection_paddingLeft};
  padding-right: ${props.removable ? "0" : props.styles_selection_paddingRight};

  & .remove:not(svg):not(path) {
    display: flex;
    border: none;
    border-radius: ${props.styles_borderRadius};
    margin-top: -${props.styles_selection_paddingTop};
    margin-bottom: -${props.styles_selection_paddingBottom};
    padding-top: ${props.styles_selection_paddingTop};
    padding-bottom: ${props.styles_selection_paddingBottom};
    padding-left: ${props.styles_selection_paddingLeft};
    padding-right: ${props.styles_selection_paddingRight};
    background-color: transparent;
    font-size: 1em;

    svg {
      fill: ${props.styles_colors_primary};
      width: ${props.styles_icon_width};
      height: ${props.styles_icon_width};
    }

    ${
      props.styles_selectionHighlighted === props.item.value
        ? `background-color: ${props.styles_colors_warning};

      svg {
        fill: ${props.styles_colors_warningBold};
      }`
        : ``
    }
  }

  ${
    props.styles_multiple
      ? `display: inline-flex;
        background-color: ${props.styles_colors_secondary};
        margin-right: ${props.styles_selection_margin};`
      : `display: flex;`
  }
  ${
    props.styles_rightToLeft
      ? `
        transform: scaleX(-1);
        flex-direction: row-reverse;
        padding-left: ${
          props.removable ? "0" : props.styles_selection_paddingRight
        };
        padding-right: ${props.styles_selection_paddingLeft};`
      : ``
  }
`
);

const Option = styled(Item)(
  (props) => `
  color: ${props.styles_colors_text}
  padding-top: ${props.styles_option_paddingTop};
  padding-bottom: ${props.styles_option_paddingBottom};
  padding-right: ${props.styles_option_paddingRight};
  padding-left: ${props.styles_option_paddingLeft};
  &:first-of-type{
    border-top-left-radius: ${props.styles_borderRadius};
    border-top-right-radius: ${props.styles_borderRadius};
  }
  &:last-of-type{
    border-bottom-left-radius: ${props.styles_borderRadius};
    border-bottom-right-radius: ${props.styles_borderRadius};
  }
  z-index: 999;

  ${
    props.styles_optionHighlighted != null &&
    props.styles_optionHighlighted === props.item.value
      ? `background-color: ${props.styles_colors_highlight};`
      : ``
  }
  ${
    props.styles_rightToLeft
      ? `transform: scaleX(-1);
        text-align: right;`
      : ``
  }
  ${
    props.item.childGroup && props.item.value == null
      ? `color: ${props.styles_colors_secondary};
        font-size: .8em;`
      : ``
  }
  ${
    props.item.childGroup && props.item.value != null
      ? `font-weight: bold;`
      : ``
  }
  ${
    props.item.group
      ? `margin-left: ${props.item.group.split(".").length * 2}em;`
      : ``
  }
`
);

export { Selection, Option };
