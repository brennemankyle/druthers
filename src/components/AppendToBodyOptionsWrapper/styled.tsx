import React from "react";
import styled from "@emotion/styled";
import {
  StyledAppendToBodyOptionsWrapper as RawStyledAppendToBodyOptionsWrapper,
  StyledAppendToBodyOptionsWrapperProps,
} from "./AppendToBodyOptionsWrapper";
import { Styles } from "../../utils/SelectTypes";
import { createBaseOptionsWrapper } from "../Wrapper/styled";

const BaseOptionsWrapper = createBaseOptionsWrapper(
  RawStyledAppendToBodyOptionsWrapper
);

const StyledAppendToBodyOptionsWrapper = styled(BaseOptionsWrapper)(
  (props: Styles & StyledAppendToBodyOptionsWrapperProps) => `
  position: absolute;
  background-color: ${props.styles_colors_background};

  ${
    props.placeOptionsAbove
      ? `bottom: ${window.innerHeight - props.parentRect.y}px;
        left: ${props.parentRect.x}px;`
      : `top: ${props.parentRect.y + props.parentRect.height}px;
        left: ${props.parentRect.x}px;`
  }
  ${props.styles_rightToLeft ? `transform: scaleX(-1);` : ``}
  ${!props.styles_hasOptions && !props.styles_multiple ? `display: none;` : ``}
`
);

export { StyledAppendToBodyOptionsWrapper };
