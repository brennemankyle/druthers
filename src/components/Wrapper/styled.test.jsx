import React from "react";
import renderer from "react-test-renderer";
import { mockStyles, MockElement } from "../../mocks";
import { Wrapper, OptionsWrapper } from "./styled";

describe("Wrapper", () => {
  it("styles", () => {
    const wrapper = renderer.create(
      <Wrapper {...mockStyles}>
        <MockElement />
      </Wrapper>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("styles_rightToLeft = true", () => {
    const wrapper = renderer.create(
      <Wrapper {...mockStyles} styles_rightToLeft={true}>
        <MockElement />
      </Wrapper>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("OptionsWrapper", () => {
  it("styles", () => {
    const wrapper = renderer.create(
      <OptionsWrapper {...mockStyles}>
        <MockElement />
      </OptionsWrapper>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("styles_hasOptions = false and styles_multiple = false", () => {
    const wrapper = renderer.create(
      <OptionsWrapper
        {...mockStyles}
        styles_hasOptions={false}
        styles_multiple={false}
      >
        <MockElement />
      </OptionsWrapper>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
