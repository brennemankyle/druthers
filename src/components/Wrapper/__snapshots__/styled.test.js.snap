// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsWrapper styles 1`] = `
.emotion-0 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 999;
  background-color: white;
  width: 600px;
}

<div
  className="emotion-0"
>
  <div
    data-props={Object {}}
  />
</div>
`;

exports[`OptionsWrapper styles_hasOptions = false and styles_multiple = false 1`] = `
.emotion-0 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 999;
  background-color: white;
  width: 600px;
  display: none;
}

<div
  className="emotion-0"
>
  <div
    data-props={Object {}}
  />
</div>
`;

exports[`Wrapper styles 1`] = `
.emotion-0 {
  font-size: 1em;
}

<div
  className="emotion-0"
>
  <div
    data-props={Object {}}
  />
</div>
`;

exports[`Wrapper styles_rightToLeft = true 1`] = `
.emotion-0 {
  font-size: 1em;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
}

<div
  className="emotion-0"
>
  <div
    data-props={Object {}}
  />
</div>
`;
